AEPServices/Sources/ui/fullscreen/FullScreenUIHandler.swift

```diff
@@ -0,0 +1,205 @@
+/*
+ Copyright 2020 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+import UIKit
+import WebKit
+
+class FullScreenUIHandler: NSObject, WKNavigationDelegate {
+
+    private let LOG_PREFIX = "FullScreenUIHandler"
+    private let DOWNLOAD_CACHE = "adbdownloadcache"
+    private let HTML_EXTENSION = "html"
+    private let TEMP_FILE_NAME = "temp"
+
+    let fileManager = FileManager()
+
+    var isLocalImageUsed = false
+    var payload: String
+    let message: FullScreenMessageUiInterface
+    var listener: FullscreenListenerInterface?
+    var monitor: MessageMonitor
+    var webView: UIView!
+
+    init(payload: String, message: FullScreenMessageUiInterface, listener: FullscreenListenerInterface?, monitor: MessageMonitor, isLocalImageUsed: Bool) {
+        self.payload = payload
+        self.message = message
+        self.listener = listener
+        self.monitor = monitor
+        self.isLocalImageUsed = isLocalImageUsed
+    }
+
+    /// Show the message, and call the OnShow() on the listener. Will only show the message if no other
+    /// fullscreen message is showing as of then.
+    func show() {
+        if monitor.isDisplayed() {
+            Log.debug(label: self.LOG_PREFIX, "Full screen message couldn't be displayed, another message is displayed at this time")
+            return
+        }
+
+        DispatchQueue.main.async {
```

Comment:
we should add some code to the `FullscreenListenerInterface` that allows the developer to suppress showing a message.  seems like it should be added here?